sdk/python/feast/infra/online_stores/astra.py

```diff
@@ -0,0 +1,301 @@
+from abc import ABC
+from datetime import datetime
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union, Sequence
+
+from cassandra.cluster import Cluster
+from cassandra.auth import PlainTextAuthProvider
+
+from feast.repo_config import FeastConfigBaseModel, RepoConfig
+from feast import Entity, FeatureTable, FeatureView
+from feast.infra.utils.online_store_utils import _table_id, _to_naive_utc
+from feast.infra.key_encoding_utils import serialize_entity_key
+from feast.protos.feast.types.EntityKey_pb2 import EntityKey as EntityKeyProto
+from feast.protos.feast.types.Value_pb2 import Value as ValueProto
+from feast.infra.online_stores.online_store import OnlineStore
+
+from pydantic import StrictStr
+from pydantic.typing import Literal
+
+
+class AstraConfig(FeastConfigBaseModel):
+    """Online store config for Astra online store"""
+
+    type: Literal["astra"] = "astra"
+    """Online store type selector"""
+
+    secure_connect_bundle: StrictStr
+    client_id: StrictStr
+    secret_key: StrictStr
+    keyspace: StrictStr
+
+
+class AstraDBOnlineStore(OnlineStore, ABC):
+    """
+    Online feature store for Astra Cassandra Database
+    """
+
+    def __init__(self, config: RepoConfig):

```

Comment:
Just out curiosity: should OnlineStore receive a config object, since the [online helper method ](https://github.com/feast-dev/feast/blob/09f525c7144ab317a15b0c670b0381a2bbd20c79/sdk/python/feast/infra/online_stores/helpers.py#L36) `get_online_store_from_config` instantiates online store without passing a config object

Reply:
> > Note: conversation about the Astra plugin has moved to #1891.
> 
> Thanks @felixwang9817 ... I interacted here to comment directly in the code 😁

I think it's great, nice code review, I fix the issues that you mention, can you please review one more time.